AdvLoggerPkg/Library/AdvancedLoggerAccessLib/AdvancedLoggerAccessLib.c

```diff
@@ -136,6 +168,40 @@ FormatPhasePrefix (
   return (UINT16)PhaseStringLen;
 }
 
+/**
+  FormatDebugLevelPrefix
+
+  Adds a debug level indicator to the message being returned.  If debug level is recognized and specified,
+  returns the debug_level prefix in from the AdvMsgEntryPrefix, otherwise raw debug level value is returned.
+
+  @param  MessageBuffer
+  @param  MessageBufferSize
+  @param  Phase
+
+  @retval Number of characters printed
+*/
+STATIC
+UINT16
+FormatDebugLevelPrefix (
+  IN CHAR8   *MessageBuffer,
+  IN UINTN   MessageBufferSize,
+  IN UINT32  DebugLevel
+  )
+{
+  UINTN  DebugLevelStringLen;
+
+  // Add the matched debug level
+  for (UINT8 i = 0; i < sizeof (debug_levels) / sizeof (debug_levels[0]); i++) {

```

Comment:
In addition to Kun's comment:

- Name the variable something other than `i` like `Index`
- Change the type of the index variable to `UINTN` instead of `UINT8`